test/unittest/logging/LogTests.cpp

```diff
@@ -143,6 +143,174 @@ TEST_F(LogTests, multiple_verbosity_levels)
+    vector<unique_ptr<thread>> threads;
+    for (int i = 0; i < threads_number; i++)
+    {
+        threads.emplace_back(new thread([&done, &committed, &thread_wait_milliseconds] {
+            while (!done)
+            {
+                logWarning(flush_ckecks, "I'm thread " << this_thread::get_id() << " logging sample " << committed);
+                // incremented after log
+                ++committed;
+                // wait before add a new log entry
+                this_thread::sleep_for(chrono::milliseconds(thread_wait_milliseconds));
+            }
+        }));
+    }
+
+    // allow some logs to be done but not all
+    while( commited_before_flush < threads_number)
+    {
+        this_thread::sleep_for(chrono::milliseconds(wait_milliseconds));
+
+        // committed value before the Flush
+        commited_before_flush = committed;
+    } 
+
+    // Wait till the queues are empty
+    Log::Flush();
+
+    int consumed = static_cast<int>(mockConsumer->ConsumedEntries().size());
+    // Flush doesn't wait for all log entries to finish but for the logged till its called
+    // We must assert that at least commited_before_flush have been delivered
+    ASSERT_GE(consumed, commited_before_flush);
+
+    logWarning(flush_ckecks, "Flushing successful, consumed: "
+        << consumed << " commited till flush " << commited_before_flush);
+
+    done = true; // direct threads to shut-down 
+
+    for (auto& thread : threads) {
+        thread->join();
+    }
+}
+
+/*
+    'validate_multithread_flush_calls' tests if the Flush() operation:
+    + assures all log entries make to this point are consumed
+    + do not deadlocks under heavy log load. Meaning that returns when there is still non consumed entries but that
+      were filed after the current flush() call.
+    + simultaneous Flush() calls from several threads do not interfere with each other operation
+*/
+TEST_F(LogTests, validate_multithread_flush_calls)
+{
+    constexpr int working_threads_number = 5;
+    constexpr int flushing_threads_number = 3;
+    constexpr int wait_milliseconds = 100;
+    constexpr int thread_wait_milliseconds = 50;
+
+    /* note that:
+        + wait_milliseconds should be larger than thread_wait_milliseconds in order allow some logs to be done
+          in each main thread loop.
+        + thread_wait_milliseconds should share order of magnitude with the consumer retrieval delay. If not so,
+          the back queue will grow so large while the front one is cleared, that the test will probably timeout.
+    */
+
+    volatile bool done = false;
+    // std::atomic<int> committed = 0; // only works on msvc and icc
+    std::atomic<int> committed;
+    committed = 0;
+
+    // Populate the consumer from multiple threads
+    vector<unique_ptr<thread>> threads;
+    for (int i = 0; i < working_threads_number; i++)
+    {
+        threads.emplace_back(new thread([&done, &committed, &thread_wait_milliseconds] {
```

Comment:
Just a suggestion. I think it is clearer 
```suggestion
        threads.emplace_back(new thread([&done, &committed, &thread_wait_milliseconds]
        {
```